### DISCUSSION TITLE
How to run clangd manually beyond clangd --check

### DISCUSSION THREAD

**Discussion by Swetko** [clangd/clangd#1215] [Q&A]
Hello, I want to play with clangd directly (The reason is that I am annoyed a bit by how it works in my editor, QtCreator, and want to see if the fault is with clangd or the editor). Is there some tutorial where it shows how you send messages and how the server responds (the format of the messages and the transport protocol). I remember seeing something about json files on stdin but when I sent some commands I didn't get any answer.

**Comment by HighCommander4** [comment]
The format of the messages is described here: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/

In terms of playing around with it, rather than typing out the JSON messages directly on the command line, I would suggest scripting it with something like Python. There is a project here with some infrastructure to do that, and an example python file that interacts with clangd: https://github.com/simark/ls-interact
